src/custom/components/SearchModal/CurrencySearch/useTokenSearch.ts

```diff
@@ -0,0 +1,75 @@
+import { useWalletInfo } from '@cow/modules/wallet'
+import { Token } from '@uniswap/sdk-core'
+import { atom, useAtom, useAtomValue } from 'jotai'
+import { loadable } from 'jotai/utils'
+import { useEffect, useMemo } from 'react'
+
+const BASE_URL = 'https://temp.api.uniswap.org/v1/tokens/search?tokenQuery='
+
+type Address = `0x${string}`
+
+interface FetchTokensResult {
+  chainId: number
+  name: string
+  address: Address
+  decimals: number
+  symbol: string
+  logoURI: string
+  coinGeckoId: string
+  priceUsd: number
+  price24hChange: number
+  volume24h: number
+  marketCap: number
+}
+
+interface FetchTokensApiResult {
+  code: number
+  status: 'success' | 'failed'
+  hasNext: boolean
+  data: FetchTokensResult[]
+}
+
+async function fetchTokens(query: string): Promise<FetchTokensApiResult> {
+  const response = await fetch(`${BASE_URL}${query}`)
+
+  return await response.json()
+}
```

Comment:
In principle,  all of the above lines could be isolated in the API layer.

Most of the new code should go to `cow-react`.

<img width="577" alt="image" src="https://user-images.githubusercontent.com/2352112/227593302-871b1d57-56c0-4b33-a212-e16de5e06cc8.png">

Reply:
Good point - I'll follow up on this to move it there.